src/server/rpc/procedures/air-quality/air-quality.js

```diff
@@ -9,94 +9,106 @@
+
+    rawLocations.pop();  // Remove trailing \n
+    rawLocations.shift();  // Remove header
+    return rawLocations
+        .map(function(line) {
+            var data = line.split('|');
+            return {
+                city: data[0],
+                state: data[1],
+                zipcode: data[2],
+                latitude: +data[3],
+                longitude: +data[4]
+            };
+        });
+})();
+
+/**
+ * Get ZIP code of closest reporting location for coordinates
+ * @param {Latitude} latitude latitude of location
+ * @param {Longitude} longitude Longitude of location
+ * @returns {Number} ZIP code of closest location
+ */
+AirConsumer._getClosestReportingLocation = function(latitude, longitude) {
+    var nearest = geolib.findNearest({latitude: latitude, longitude: longitude}, reportingLocations),
         city = reportingLocations[nearest.key].city,
         state = reportingLocations[nearest.key].state,
         zipcode = reportingLocations[nearest.key].zipcode;
-    logger.trace('Nearest reporting location is ' + city + ', ' + state);
+    trace('Nearest reporting location is ' + city + ', ' + state);
     return zipcode;
 };
 
-var qualityIndex = function(latitude, longitude) {
-    var response = this.response,
-        nearest,
-        url;
+/**
+ * Get air quality index of closest reporting location for coordinates
+ * @param {Latitude} latitude latitude of location
+ * @param {Longitude} longitude Longitude of location
+ * @returns {Number} AQI of closest station
+ */
+AirConsumer.qualityIndex = function(latitude, longitude) {
+    var nearest = this._getClosestReportingLocation(latitude, longitude);
 
-    logger.trace(`Requesting air quality at ${latitude}, ${longitude}`);
-    if (!latitude || !longitude) {
-        return response.status(400).send('ERROR: missing latitude or longitude');
-    }
+    trace(`Requesting air quality at ${latitude}, ${longitude} (nearest station: ${nearest})`);
 
-    nearest = getClosestReportingLocation(latitude, longitude);
-    url = baseUrl + '&zipCode=' + nearest;
-
-    logger.trace('Requesting air quality at '+ nearest);
-    
-    request(url, (err, res, body) => {
-        var aqi = -1,
-            code = err ? 500 : res.statusCode;
-
-        try {
-            body = JSON.parse(body).shift();
-            if (body && body.AQI) {
-                aqi = +body.AQI;
-                logger.trace('Air quality at '+ nearest + ' is ' + aqi);
-            }
-        } catch (e) {
-            // Just send -1 if anything bad happens
-            logger.error('Could not get air quality index: ', e);
-        }
+    return this.qualityIndexByZip(nearest);
+};
 
-        response.status(code).json(aqi);
-    });
+/**
+ * Get air quality index of closest reporting location for ZIP code
+ * @param {BoundedNumber<0,99999>} zipCode ZIP code of location
+ * @returns {Number} AQI of closest station
+ */
+AirConsumer.qualityIndexByZip = function(zipCode) {
```

Comment:
Did you mean to add another method? If not, prefix the function name with a `_`. If so, the name should maybe be `qualityIndexByZipCode` rather than just `Zip` (some of our users can be young and we want to make it as obvious as possible.

Reply:
It's intended, name is changed.